tests/lava/magma/core/process/ports/test_virtual_ports_in_process.py

```diff
@@ -40,15 +40,15 @@ class MockVirtualPort(AbstractVirtualPort, AbstractPort):
     """A mock-up of a virtual port that reshapes the input."""
     def __init__(self,
                  new_shape: ty.Tuple[int, ...],
-                 old_shape: ty.Optional[ty.Tuple[int, ...]] = None):
+                 axes: ty.Tuple[int, ...]):
         AbstractPort.__init__(self, new_shape)
-        self.old_shape = old_shape
+        self.axes = axes
 
     def get_transform_func_fwd(self) -> ft.partial:
```

Comment:
Is the reason we need this as the Port level because we need it for both PyPorts and NcPorts?

Is this actually just needed in a PyPort or do we also need it in the compiler to sort indices?

Reply:
That is not yet clear to me, it would depend on how I implement the index remapping for NcPorts. I initially saw this as specific to PyPorts.
At the moment it lives there because virtual ports never had dedicated implementations for different backends.

I agree though - if the implementation of NcPorts turns out not to depend on these function pointers, I would split up the implementation of virtual ports into versions dedicated to Nc, Py, etc.